cxx-qt-lib/src/lib.rs

```diff
@@ -49,71 +49,262 @@ pub use qvariant::{QVariant, Variant, VariantValue};
 mod map_qt_value;
 pub use map_qt_value::*;
 
-pub const QT_TYPES_HEADER: &str = include_str!("../include/qt_types.h");
+// Provide a separate depending on the platform
+// this is because include_str requires th correct and non-mixed path separators
+//
+// https://github.com/rust-lang/rust/issues/75075
+#[cfg(not(windows))]
+macro_rules! sep {
+    () => {
+        "/"
+    };
+}
+
+#[cfg(windows)]
+macro_rules! sep {
+    () => {
+        "\\"
+    };
+}
```

Comment:
Macros seem overcomplicated here. Can't you use [std](https://doc.rust-lang.org/stable/std/index.html)::[path](https://doc.rust-lang.org/stable/std/path/index.html)::[MAIN_SEPARATOR](https://doc.rust-lang.org/stable/std/path/constant.MAIN_SEPARATOR.html#)? Or failing that, conditionally define a static &str like https://github.com/RustAudio/rust-jack/blob/main/jack-sys/src/lib.rs#L6

Reply:
Me, but for Windows \ path separators
https://xkcd.com/567/

Reply:
Only string literals can be used with `concat!` and `include_str!` so `MAIN_SEPARATOR` cannot be used. The macro could maybe be improved so we do't need to put `sep!` constantly, i'll look at this later.